src/app/shared/date-time/picker/date-time-picker.component.ts

```diff
@@ -0,0 +1,183 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {Overlay, OverlayRef} from '@angular/cdk/overlay';
+import {Portal, TemplatePortal} from '@angular/cdk/portal';
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnDestroy,
+  Output,
+  SimpleChanges,
+  TemplateRef,
+  ViewChild,
+  ViewContainerRef,
+} from '@angular/core';
+import {AbstractControl, FormControl, FormGroup} from '@angular/forms';
+import * as moment from 'moment';
+import {BsDatepickerInlineConfig, BsLocaleService} from 'ngx-bootstrap/datepicker';
+import {environment} from '../../../../environments/environment';
+import {DateTimeOptions, detectDatePickerViewMode} from '../date-time-options';
+
+@Component({
+  selector: 'date-time-picker',
+  templateUrl: './date-time-picker.component.html',
+  styleUrls: ['./date-time-picker.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DateTimePickerComponent implements OnChanges, AfterViewInit, OnDestroy {
+  @Input()
+  public origin: ElementRef | HTMLElement;
+
+  @Input()
+  public min: Date;
+
+  @Input()
+  public max: Date;
+
+  @Input()
+  public options: DateTimeOptions;
+
+  @Input()
+  public value: Date;
+
+  @Output()
+  public valueChange = new EventEmitter<Date>();
+
+  @ViewChild('dateTimePicker')
+  public dateTimePicker: TemplateRef<any>;
+
+  public form = new FormGroup({
+    date: new FormControl(),
+  });
+
+  public datePickerConfig: Partial<BsDatepickerInlineConfig>;
+
+  public timeZone = `UTC${moment().format('Z')}, ${Intl.DateTimeFormat().resolvedOptions().timeZone}`;
+
+  private overlayRef: OverlayRef;
+  private portal: Portal<any>;
+
+  constructor(localeService: BsLocaleService, private overlay: Overlay, private viewContainer: ViewContainerRef) {
+    localeService.use(environment.locale);
+  }
+
+  public ngOnChanges(changes: SimpleChanges) {
+    if (changes.value && this.value) {
+      this.dateControl.setValue(this.value);
+    }
+    if (changes.options && this.options) {
+      this.datePickerConfig = {
+        containerClass: 'box-shadow-none theme-default',
+        minMode: detectDatePickerViewMode(this.options),
+      };
+    }
+  }
+
+  public ngAfterViewInit() {
+    this.portal = new TemplatePortal(this.dateTimePicker, this.viewContainer);
+  }
+
+  public ngOnDestroy() {
+    this.close();
+  }
+
+  public onClick(event: MouseEvent) {
+    // otherwise it is immediately closed in table
+    event.stopPropagation();
+  }
+
+  public open() {
+    if (this.overlayRef) {
+      return;
+    }
+
+    this.overlayRef = this.overlay.create({
```

Comment:
Interesting! Can you tell me something about it on todays meeting?

Reply:
Yes, but it is just normal [Overlay](https://material.angular.io/cdk/overlay/overview) from Angular CDK.